Include/cpython/object.h

```diff
@@ -263,6 +263,10 @@ struct _typeobject {
 
     destructor tp_finalize;
     vectorcallfunc tp_vectorcall;
+
+    /* INTERNAL USE ONLY! MODIFYING THIS CAN CRASH PYTHON! */
+    const Py_ssize_t tp_obj_offset; /* Offset from "PyObject *" pointer */
```

Comment:
I don't think that this is needed. When I get a "PyObject *ob", the offset to the PyObject structure is 0: "PyObject copy = *ob;" in Python internals is valid.

Reply:
It's not required if everything inherited ONLY PyObject, that will never be true.

The offset is calculated from the immediate base type of the type, which can be for example: list, tuple, dict, type, custom types.

Same thing applies to "tp_obj_size", without these members, heap corruption can be very likely.